fem/gridfunc.hpp

```diff
@@ -141,17 +141,134 @@ class GridFunction : public Vector
    /// Returns the values in the vertices of i'th element for dimension vdim.
    void GetNodalValues(int i, Array<double> &nval, int vdim = 1) const;
 
+   /** @name Element index Get Value Methods
+
+       These methods take an element index and return the interpolated
+       value of the field at a given reference point within the
+       element.
+   */
+   ///@{
+   /** Return a scalar value from within the given element. */
```

Comment:
Maybe add doxygen warning (`@warning`) or note (`@note`) about possible internal clashes we have encountered in #716, because that behavior is very unexpected (at least for users not familiar with inner working of MFEM :smile:)

Reply:
Good idea!